appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/CommitDelayingResponse.java

```diff
@@ -77,9 +77,6 @@
    */
   public CommitDelayingResponse(HttpServletResponse response) throws IOException {
     super(response);
-    // Set the buffer size explicitly, as this ensure both that jetty's aggregate
-    // buffer is the correct size and that the commit threshold is that size.
-    response.setBufferSize(CommitDelayingOutputStream.MAX_RESPONSE_SIZE_BYTES);
```

Comment:
By default the output buffer is 32KB and the commit write threshhold is 8KB.  Note that this means that if first write is larger than 8KB in a single call to write, then it will be directly written (and a commit done).    But if the initial write is < 8KB then we allocate a 32KB aggregate buffer and will not commit until it is full (or flush is called).

So there are probably few apps that write dynamic content in chunks larger than 8KB.

Jetty does not use direct buffers for aggregate buffers, because they are written too many times and flushed once - hence they are best kept in user space.    We only use direct buffers for caching large content that cannot be file mapped and is written in a single call (and then reused).

We pool heap buffers.